newIDE/app/src/Utils/UseCreateProject.js

```diff
@@ -215,13 +226,29 @@ const useCreateProject = ({
             }
           );
 
-          if (wasSaved) {
-            onProjectSaved(fileMetadata);
-            unsavedChanges.sealUnsavedChanges({ setCheckpointTime: true });
-            if (newProjectSetup.storageProvider.internalName === 'LocalFile') {
-              preferences.setHasProjectOpened(true);
-            }
+          if (!wasSaved) {
+            return; // Saving was cancelled.
+          }
+
+          if (!fileMetadata) {
+            return;
           }
+
+          onProjectSaved(fileMetadata);
+          unsavedChanges.sealUnsavedChanges({ setCheckpointTime: true });
+          if (newProjectSetup.storageProvider.internalName === 'LocalFile') {
+            preferences.setHasProjectOpened(true);
+          }
+
+          // Save was done on a new file/location, so save it in the
+          // recent projects and in the state.
+          const fileMetadataAndStorageProviderName = {
+            fileMetadata,
+            storageProviderName: storageProviderInternalName,
+          };
+          preferences.insertRecentProjectFile(
```

Comment:
this is the fix to show the project in the recent files as soon as it is created with a chosen storage provider

Reply:
👍